pkg/retryablehttp/httpclient.go

```diff
@@ -0,0 +1,75 @@
+package retryablehttp
+
+import (
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+)
+
+const backOffMaxDuration = 3 * time.Second
+
+type RetryableRoundTripper struct {
+	Client *RetryableHTTPClient
+	once   sync.Once
+}
+
+// RoundTripper is an interface representing the ability to execute a single HTTP transaction, obtaining the Response for a given Request.
+// A RoundTripper must be safe for concurrent use by multiple goroutines.
+var _ http.RoundTripper = (*RetryableRoundTripper)(nil)
+
+func (rt *RetryableRoundTripper) init() {
+	if rt.Client == nil {
+		rt.Client = NewRetryableHTTPClient()
+	}
+}
+
+// RoundTrip executes a single HTTP transaction, but does not attempt to read the response, modify it, or close the body.
+func (rt *RetryableRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	rt.once.Do(rt.init)
+	return rt.Client.Do(req)
+}
+
+type RetryableHTTPClient struct {
+	internalClient http.Client
+}
+
+func NewRetryableHTTPClient() *RetryableHTTPClient {
```

Comment:
Small nit: but do you mind calling this `NewClient`, or something similar, as this is already the `retryablehttp` package?